src/demo/components/Graphs.tsx

```diff
@@ -0,0 +1,230 @@
+import '../demo.css'
+
+import React from 'react'
+import {
+  Bar,
+  CartesianGrid,
+  ComposedChart,
+  Line,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis
+} from 'recharts'
+
+interface Bucket {
+  start: number
+  usdValue: number
+  numTxs: number
+  isoDate: string
+  currencyCodes: { [currencyCode: string]: number }
+  currencyPairs: { [currencyPair: string]: number }
+}
+
+interface AnalyticsResult {
+  result: {
+    hour: Bucket[]
+    day: Bucket[]
+    month: Bucket[]
+    numAllTxs: number
+  }
+  app: string
+  pluginId: string
+  start: number
+  end: number
+}
+
+const Graphs: any = (props: {
+  rawData: AnalyticsResult[]
+  timePeriod: string
+  colors: string[]
+}) => {
+  const { rawData, timePeriod } = props
+  const keys = props.rawData.map(analyticsResult => {
+    return (
+      analyticsResult.pluginId.charAt(0).toUpperCase() +
+      analyticsResult.pluginId.slice(1)
+    )
+  })
+  const colors = {}
+  for (let i = 0; i < rawData.length; i++) {
+    colors[keys[i]] = props.colors[i]
+  }
+  const tickRate: string[] = []
+  let tickSpace = 0
+  tickSpace = Math.floor(rawData[0].result[timePeriod].length / 5)
+  if (tickSpace === 0) tickSpace++
+
+  const data = rawData[0].result[timePeriod].map((bucket: Bucket, index) => {
+    let date
+    if (timePeriod === 'month') {
+      date = bucket.isoDate.slice(0, 7)
+    } else if (timePeriod === 'day') {
+      date = bucket.isoDate.slice(0, 10)
+    } else {
+      date = bucket.isoDate.slice(0, 10) + ':' + bucket.isoDate.slice(11, 13)
+    }
+    if (index % tickSpace === 0) {
+      tickRate.push(date)
+    }
+    const currencyPairs: { [currencyPair: string]: number } = {}
+    for (const bucket of rawData) {
```

Comment:
for (const bucket of rawData) {
      const { currencyPairs } = bucket.result[timePeriod][index]
      for (const key in currencyPairs) {
        const value = currencyPairs[key]
        if (typeof value === 'number') {
          if (currencyPairs[key] == null) {
            currencyPairs[key] = value
            continue
          }
          currencyPairs[key] += value
        }
      }
    }